java/src/main/java/com/google/crypto/tink/integration/awskms/AwsKmsClient.java

```diff
@@ -110,28 +110,29 @@ public KmsClient withDefaultCredentials() throws GeneralSecurityException {
   }
 
   /** Loads AWS credentials from a provider. */
-  public KmsClient withCredentialsProvider(AWSCredentialsProvider provider)
-      throws GeneralSecurityException {
-    try {
-      String[] tokens = this.keyUri.split(":");
-      this.client =
-          AWSKMSClientBuilder.standard()
-              .withCredentials(provider)
-              .withRegion(Regions.fromName(tokens[4]))
-              .build();
+    public KmsClient withCredentialsProvider(AWSCredentialsProvider provider)
+        throws GeneralSecurityException {
+      this.provider = provider;
       return this;
-    } catch (AmazonServiceException e) {
-      throw new GeneralSecurityException("cannot load credentials from provider", e);
     }
-  }
-
-  @Override
-  public Aead getAead(String uri) throws GeneralSecurityException {
-    if (this.keyUri != null && !this.keyUri.equals(uri)) {
-      throw new GeneralSecurityException(
-          String.format(
-              "this client is bound to %s, cannot load keys bound to %s", this.keyUri, uri));
+    @Override
+    public Aead getAead(String uri) throws GeneralSecurityException {
+      if (this.keyUri != null && !this.keyUri.equals(uri)) {
+        throw new GeneralSecurityException(
+            String.format(
+                "this client is bound to %s, cannot load keys bound to %s", this.keyUri, uri));
+      }
+    try {
+      String keyUri = Validators.validateKmsKeyUriAndRemovePrefix(PREFIX, uri);
+      String[] tokens = keyUri.split(":");
+      AWSKMS client =
+            AWSKMSClientBuilder.standard()
+                .withCredentials(provider)
+                .withRegion(Regions.fromName(tokens[4]))
```

Comment:
Are you sure the region is in tokens[4]?

Our test key is arn:aws:kms:us-east-2:235739564943:key/3ee50705-5a82-4f5b-9753-05c4f473922f. tokens[4] is 235739564943. I think it is token[3]?

BTW you can test this yourself, by running

cd tools/
bazel test/testing/cross_language/...